internal/cmd/stencil/stencil.go

```diff
@@ -152,31 +150,72 @@ func (c *Command) useModulesFromLock() error {
 		return fmt.Errorf("frozen lockfile requires a lockfile to exist")
 	}
 
+	desiredModulesHM := make(map[string]bool)
 	for _, m := range c.manifest.Modules {
-		// Convert m.URL -> m.Name
-		//nolint:staticcheck // Why: We're implementing compat here.
-		if m.URL != "" {
-			u, err := giturls.Parse(m.URL) //nolint:staticcheck // Why: We're implementing compat here.
-			if err != nil {
-				//nolint:staticcheck // Why: We're implementing compat here.
-				return errors.Wrapf(err, "failed to parse deprecated url module syntax %q as a URL", m.URL)
-			}
-			m.Name = path.Join(u.Host, u.Path)
+		desiredModulesHM[m.Name] = true
+	}
+
+	lockfileModulesHM := make(map[string]*stencil.LockfileModuleEntry)
+	for _, m := range c.lock.Modules {
+		lockfileModulesHM[m.Name] = m
+	}
+
+	outOfSync := false
+	outOfSyncReasons := make([]string, 0)
+
+	// iterate over all of the modules that are desired, if
+	// they are not in the lockfile, then the user is unable
+	// to use a frozen lockfile.
+	for _, m := range c.manifest.Modules {
+		if _, ok := lockfileModulesHM[m.Name]; !ok {
+			outOfSync = true
+			outOfSyncReasons = append(outOfSyncReasons,
+				fmt.Sprintf("module %s requested by service.yaml but is not in the lockfile", m.Name))
+			break
```

Comment:
Why do you break here? Don't you want to gather all reasons?

Reply:
Because Jesse, you're not the boss of me. o wait. No, in reality I changed this to gather multiple reasons and never removed the break. That's why :D

Reply:
also seems like an opportunity for your unit tests to cover this case.

Reply:
Github showed me this after merge :sob:

Since it's just logged right now I'd have to change the log messages, but yeah this is potentially something we could unit test. I feel _generally_ safe enough with it checking error cases right now though